src/transformers/trainer.py

```diff
@@ -402,19 +402,23 @@ def __init__(
                     " to `True` to avoid any unexpected behavior such as device placement mismatching."
                 )
 
+        _is_peft_model = is_peft_available() and isinstance(model, PeftModel)
+        _is_quantized_and_base_model = getattr(model, "is_quantized", False) and not getattr(
+            model, "_hf_peft_config_loaded", False
+        )
+
         # At this stage the model is already loaded
-        if getattr(model, "is_quantized", False) and not getattr(model, "_hf_peft_config_loaded", False):
-            if getattr(model, "_is_quantized_training_enabled", False):
-                logger.info(
-                    "The model is quantized. To train this model you need to add additional modules"
-                    " inside the model such as adapters using `peft` library and freeze the model weights. Please"
-                    " check the examples in https://github.com/huggingface/peft for more details."
-                )
-            else:
-                raise ValueError(
-                    "The model you want to train is loaded in 8-bit precision.  if you want to fine-tune an 8-bit"
-                    " model, please make sure that you have installed `bitsandbytes>=0.37.0`. "
-                )
+        if _is_quantized_and_base_model and not _is_peft_model:
```

Comment:
Not sure about this new condition `not _is_peft_model`. I think that having `not getattr(model, "_hf_peft_config_loaded", False)` implies `not _is_peft_model` . Is there a case where the peft config is not loaded but the model is a peft model ? What will happen is that if we have `_is_quantized_and_base_model`, only the first error will be triggered.

Reply:
What I wanted to avoid here is that if you pass a PeftModel it will still contain the attributes of the base model - including `is_quantized` etc.): https://github.com/huggingface/peft/blob/main/src/peft/peft_model.py#L456 - so that check circumvents the case where you pass a pure quantized model, with no adapter attached, and not being a PeftModel

Reply:
Makes sense. Thanks for explaining !